src/Microsoft.Graph.Core/Requests/Middleware/AuthenticationHandler.cs

```diff
@@ -131,5 +141,47 @@ protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage
                 return await base.SendAsync(httpRequestMessage, cancellationToken);
             }
         }
+
+        /// <summary>
+        /// Add claims to the request context of the given request message
+        /// </summary>
+        /// <param name="wwwAuthenticateHeader">String representation of www Authenticate Header</param>
+        /// <param name="newRequest">Request message to add claims to</param>
+        private void AddClaimsToRequestContext(HttpRequestMessage newRequest, string wwwAuthenticateHeader)
+        {
+            int claimsStart = wwwAuthenticateHeader.IndexOf("claims=", StringComparison.OrdinalIgnoreCase);
+            if (claimsStart < 0) 
+                return; // do nothing as there is no claims in www Authenticate Header
+
+            claimsStart += 8; // jump to the index after the opening quotation mark
```

Comment:
can you include a sample of what the string here looks like.  Is this likely to remain stable?

Reply:
I have tried to capture these in the tests.

Reply:
What in the `www-Authenticate` [specification](https://www.rfc-editor.org/rfc/rfc7235.html#section-4.1) leads you to believe that the challenge parameter "claims" is necessarily the last parameter?

Reply:
This is resolved in the latest versions of the SDK via https://github.com/microsoftgraph/msgraph-sdk-dotnet-core/pull/459/files